src/matrix.js

```diff
@@ -82,21 +179,47 @@ SVG.Matrix = SVG.invent({
 
       return matrix
     },
-    // Multiplies by given matrix
+
+    // Left multiplies by the given matrix
     multiply: function (matrix) {
-      return new SVG.Matrix(this.native().multiply(parseMatrix(matrix).native()))
+      // Get the matrices
+      var l = this
+      var r = parseMatrix(matrix)
+
+      // Work out the product directly
+      var a = l.a * r.a + l.c * r.b
+      var b = l.b * r.a + l.d * r.b
+      var c = l.a * r.c + l.c * r.d
+      var d = l.b * r.c + l.d * r.d
+      var e = l.e + l.a * r.e + l.c * r.f
+      var f = l.f + l.b * r.e + l.d * r.f
+
+      // Form the matrix and return it
+      var product = new SVG.Matrix(a, b, c, d, e, f)
+      return product
```

Comment:
So I guess native is completely gone? :D

Reply:
I actually wonder which is faster

Reply:
I think it was done to reduce filesize and potential errors

Reply:
Yea its not really necessary to do this. I could put it back. But it just avoids having to parse both matrices twice.